File: Arrays and Strings/1.2 - Check Permutation/main.py
def isAnagram(stringOne, stringTwo):
    
    letterTable = {}

    for n in range( len(stringOne) ):
        letterTable[stringOne[n]] = 1 + letterTable.get(stringOne[n], 0)
        letterTable[stringTwo[n]] = -1 + letterTable.get(stringTwo[n], 0)

    for val in letterTable:
        if letterTable[val] > 0:
            return False

    return True

#d[key] = d.get(key, value)


one = "apple"
two = "pplse"

value = isAnagram(one, two)

print("Is permutation?: ")
print(value)

File: Arrays and Strings/1.2 - Check Permutation/C++/main.cpp
#include <iostream>
#include <string>
#include <unordered_map>
using namespace std;

bool isAnagram(string StringOne, string StringTwo);

int main(){

    string s1 = "apples";
    string s2 = "aaples";

    if( isAnagram(s1, s2) ){
        printf("\nThe Strings are anagrams!\n");
    } else {
        printf("\nThe Strings are not anagrams :(\n");
    }

    return 0;

}

bool isAnagram(string StringOne, string StringTwo){

    unordered_map<char, int> table;

    if(StringOne.size() != StringTwo.size()){
        return false;
    }

    for(int i=0; i < StringOne.size(); ++i){
        table[ StringOne[i] ]++;
        table[ StringTwo[i] ]--;
    }

    for( auto i : table){

        if(i.second > 0){
            return false;
        }

    }


    return true;
}